kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -154,11 +154,27 @@ protected static ByteBuf responseToByteBuf(AbstractResponse response, KafkaHeade
         }
     }
 
+    protected Boolean channelReady() {
+        return hasAuthenticated();
+    }
+
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
         // Get a buffer that contains the full frame
         ByteBuf buffer = (ByteBuf) msg;
 
+        if (isActive.get() && !channelReady()) {
+            try {
+                channelPrepare(ctx, buffer);
+                return;
+            } catch (AuthenticationException e) {
+                log.error("unexpected error in authenticate:", e);
+                close();
+            } finally {
+                buffer.release();
+            }
+        }
```

Comment:
When the channel receives a request, it needs to confirm whether the current channel is authenticated, otherwise, execute channelPrepare and complete the authentication after multiple interactions.